src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -20832,6 +20837,267 @@ static void Report<T>(MyCollection<T> c)
+                {
+                  // Code size       36 (0x24)
+                  .maxstack  2
+                  .locals init (System.Span<int> V_0, //y
+                                int V_1)
+                  IL_0000:  ldarg.0
+                  IL_0001:  stloc.1
+                  IL_0002:  ldloca.s   V_1
+                  IL_0004:  newobj     "System.Span<int>..ctor(ref int)"
+                  IL_0009:  stloc.0
+                  IL_000a:  ldarg.0
+                  IL_000b:  ldc.i4.1
+                  IL_000c:  add
+                  IL_000d:  starg.s    V_0
+                  IL_000f:  ldloca.s   V_0
+                  IL_0011:  ldc.i4.0
+                  IL_0012:  call       "ref int System.Span<int>.this[int].get"
+                  IL_0017:  ldind.i4
+                  IL_0018:  call       "void System.Console.Write(int)"
+                  IL_001d:  ldarg.0
+                  IL_001e:  call       "void System.Console.Write(int)"
+                  IL_0023:  ret
+                }
+                """);
+
+            verifier = CompileAndVerify(source, targetFramework: TargetFramework.Net70, verify: Verification.Skipped, expectedOutput: IncludeExpectedOutput("12"));
+            verifier.VerifyDiagnostics();
+
+            verifier.VerifyIL("Program.M", """
+                {
+                  // Code size       43 (0x2b)
+                  .maxstack  5
+                  .locals init (System.Span<int> V_0) //y
+                  IL_0000:  ldloca.s   V_0
+                  IL_0002:  ldc.i4.1
+                  IL_0003:  newarr     "int"
+                  IL_0008:  dup
+                  IL_0009:  ldc.i4.0
+                  IL_000a:  ldarg.0
+                  IL_000b:  stelem.i4
+                  IL_000c:  call       "System.Span<int>..ctor(int[])"
+                  IL_0011:  ldarg.0
+                  IL_0012:  ldc.i4.1
+                  IL_0013:  add
+                  IL_0014:  starg.s    V_0
+                  IL_0016:  ldloca.s   V_0
+                  IL_0018:  ldc.i4.0
+                  IL_0019:  call       "ref int System.Span<int>.this[int].get"
+                  IL_001e:  ldind.i4
+                  IL_001f:  call       "void System.Console.Write(int)"
+                  IL_0024:  ldarg.0
+                  IL_0025:  call       "void System.Console.Write(int)"
+                  IL_002a:  ret
+                }
+                """);
+        }
+
+        [Fact]
+        public void Span_SingleElement_TempReuse()
+        {
+            var source = """
+                using System;
+
+                class Program
+                {
+                    static void Main() => M(1);
+
+                    static void M(int x)
+                    {
+                        {
+                            Span<int> y = [x];
+                            Console.Write(y[0]);
+                        }
+                        {
+                            Span<int> y = [x];
```

Comment:
Do we have a test where we create single-element collection twice in the same block (not different blocks like in this test)?

Reply:
Indeed this is the type of test that I was most worried about. Re-using the temp between two different blocks is just fine. Re-using it within the same block would be a significant codegen error. Want to make sure that we're not doing that.